modules/reference/pages/config/server-configuration-overview.adoc

```diff
@@ -194,6 +195,25 @@ For example, if you enter `${my.env.var}` in the `server.xml` file, it looks for
 . my_env_var
 . MY_ENV_VAR
 
+Variables can be loaded from files located in directories that are specified by the `VARIABLE_SOURCE_DIRS` environment variable.
+If this environment variable is not specified, the default location for the variable directory is `${server.config.dir}/variables`.
+
+The name of the file is used as the variable name, and the contents of the file are used as the variable value.
+For example, you can create a file in the variables directory that is named `httpPort` with the text string `9080` as the only content.
+You can then use the variable `${httpPort}` in the `server.xml` file and it resolves to `9080`.
+
+If you create a file in a directory within the variables directory, then the parent directory is added to the variable name.
+For example, you can create a `${httpPort}` file in the `ports` subdirectory within the variable directory, the variable name is `${ports/httpPort}`.
+
+Variables in the `VARIABLE_SOURCE_DIRS` directories can also be defined in the properties files.
```

Comment:
Variables in the `VARIABLE_SOURCE_DIRS` directories can also be defined in the properties files.
--->
Variables in the `VARIABLE_SOURCE_DIRS` directories can also be defined in properties files.